### PR TITLE
CLI documentation page generation

### PR DESCRIPTION
**Pull request by Lilferrit**
I implemented automatic documentation generation for the casanovo CLI using the sphinx-click package. This does introduce a new dependency on the sphinx-click python package in order to build the documentation.

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Casanovo CLI
+==============
```
Comment (bittremieux) [inline docs/cli.rst:2]: **nitpick:** Align the "=" to the text.

File: docs/cli.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+After installing the Casanovo package and all of its dependencies, the Casanovo CLI
```
Comment (bittremieux) [inline docs/cli.rst:4]: **suggestion:** For nicer git diffs, in text files you can split at the sentence level.

File: docs/_static/js/cli-page.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+(function () {
```
Comment (bittremieux) [inline docs/_static/js/cli-page.js:1]: **question:** Do we need this Javascript hack? How does it look without/with doing this?

File: docs/images/configure-help.svg
Comment (bittremieux) [inline docs/images/configure-help.svg:1]: **todo:** Merge in the latest `dev` to get proper screenshots.

**Review by bittremieux — COMMENTED** [review]
Looks fine, just a few small clarifications and the latest changes on `dev` need to be merged in.

File: docs/_static/js/cli-page.js (hunk 3 above)
Comment (Lilferrit) [reply docs/_static/js/cli-page.js:1]: Unfortunately the due to the way that the sphinx-click package is implemented the generated CLI documentation page looks pretty terrible without it.

Comment (bittremieux) [reply docs/_static/js/cli-page.js:1]: Can you share a screenshot with and without it? Doesn't it just remove a header?

Comment (bittremieux) [reply docs/_static/js/cli-page.js:1]: It's not ideal, but also not entirely terrible imo. And adding Javascript to all pages feels pretty unsatisfying as well.

Apparently there were some efforts to [remove the header with spinx-click](https://github.com/click-contrib/sphinx-click/pull/100), but they have stalled unfortunately.

Can you use [`sphinx-click-process-description`](https://sphinx-click.readthedocs.io/en/latest/usage/#events) to remove the second Casanovo header that comes from the docstring? That should bring us pretty close to the desired layout already.

Comment (Lilferrit) [reply docs/_static/js/cli-page.js:1]: I changed the implementation to use an event handler instead, here's what it looks like:

image

File: docs/conf.py [hunk 4]
```diff
@@ -12,9 +12,10 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath("."))
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("."))
```
Comment (bittremieux) [inline docs/conf.py:18]: **nitpick:** Two blank lines after the imports.

File: docs/conf.py [hunk 5]
```diff
@@ -57,9 +58,13 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ["_static"]
+# html_static_path = ["_static"]
 html_theme_options = {"repository_url": "https//github.com/Noble-Lab/casanovo"}
 
+html_js_files = [
```
Comment (bittremieux) [inline docs/conf.py:64]: **todo:** This can be removed.

File: docs/conf.py [hunk 6]
```diff
@@ -57,9 +58,13 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ["_static"]
+# html_static_path = ["_static"]
```
Comment (bittremieux) [inline docs/conf.py:61]: **question:** Does this need to be commented?

File: docs/sphinx_click_handlers.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+def process_description(app, ctx, lines):
+    if ctx.command.name == "main":
+        del lines[:3]
+
+    print(lines)
```
Comment (bittremieux) [inline docs/sphinx_click_handlers.py:5]: **question:** Does this need to be there? The sphinx-click documentation says that the content is modified _in-place_, so I don't think it needs to be printed as well.

**Review by bittremieux — CHANGES_REQUESTED** [review]
Final small comments.

File: docs/_static/js/cli-page.js (hunk 3 above)
Comment (bittremieux) [reply docs/_static/js/cli-page.js]: Yes, I think that this implementation is better. And the page doesn't look awful.

File: docs/conf.py (hunk 6 above)
Comment (Lilferrit) [reply docs/conf.py:61]: AFIK this was originally left as a commented out piece of code in the conf.py template, but I'll go ahead and remove it.

File: docs/sphinx_click_handlers.py (hunk 7 above)
Comment (Lilferrit) [reply docs/sphinx_click_handlers.py:5]: My bad, that was some debug code I forgot to remove from the commit.

File: docs/conf.py (hunk 6 above)
Comment (bittremieux) [reply docs/conf.py]: It's currently not commented on `main`, so unless the change is required, I think we can leave it as-is,

**Review by bittremieux — APPROVED** [review]

**Pull request merged**